contracts/SP3QSAJQ4EA8WXEDSRRKMZZ29NH91VZ6C5X88FGZQ.crashpunks-v1.clar

```diff
@@ -0,0 +1,1004 @@
+    (begin
+        (asserts! (is-eq (var-get administrator) contract-caller) not-allowed)
+        (var-set collection-mint-addresses new-mint-addresses)
+        (var-set collection-mint-shares new-mint-shares)
+        (var-set collection-addresses new-addresses)
+        (var-set collection-shares new-shares)
+        (var-set collection-secondaries new-secondaries)
+        (ok true)
+    )
+)
+
+(define-public (set-collection-mint-addresses (new-mint-addresses (list 4 principal)))
+    (begin
+        (asserts! (is-eq (var-get administrator) contract-caller) not-allowed)
+        (var-set collection-mint-addresses new-mint-addresses)
+        (ok true)
+    )
+)
+
+(define-public (set-collection-mint-shares (new-mint-shares (list 4 uint)))
+    (begin
+        (asserts! (is-eq (var-get administrator) contract-caller) not-allowed)
+        (var-set collection-mint-shares new-mint-shares)
+        (ok true)
+    )
+)
+
+(define-public (set-collection-addresses (new-addresses (list 10 principal)))
+    (begin
+        (asserts! (is-eq (var-get administrator) contract-caller) not-allowed)
+        (var-set collection-addresses new-addresses)
+        (ok true)
+    )
+)
+
+(define-public (set-collection-shares (new-shares (list 10 uint)))
+    (begin
+        (asserts! (is-eq (var-get administrator) contract-caller) not-allowed)
+        (var-set collection-shares new-shares)
+        (ok true)
+    )
+)
+
+(define-public (set-collection-secondaries (new-secondaries (list 10 uint)))
+    (begin
+        (asserts! (is-eq (var-get administrator) contract-caller) not-allowed)
+        (var-set collection-secondaries new-secondaries)
+        (ok true)
+    )
+)
+
+;; Transfers tokens to a 'SPecified principal.
+(define-public (transfer (nftIndex uint) (owner principal) (recipient principal))
+  (if (unwrap! (is-approved nftIndex owner) nft-not-owned-err)
+    (match (nft-transfer? crashpunks nftIndex owner recipient)
+        success (begin (ok success))
+        error (nft-transfer-err error))
+    nft-not-owned-err)
+)
+
+(define-public (transfer-memo (id uint) (sender principal) (recipient principal) (memo (buff 34)))
+    (let ((result (transfer id sender recipient)))
+      (print memo)
+      result))
+
+;; Burns tokens
+(define-public (burn (nftIndex uint) (owner principal))
+  (if (unwrap! (is-approved nftIndex owner) nft-not-owned-err)
+    (match (nft-burn? crashpunks nftIndex owner)
+        success (begin
+            (ok success)
+        )
+        error (nft-transfer-err error))
+    nft-not-owned-err)
+)
+
+(define-private (nft-transfer-err (code uint))
+  (if (is-eq u1 code)
+    nft-not-owned-err
+    (if (is-eq u2 code)
+      sender-equals-recipient-err
+      (if (is-eq u3 code)
+        nft-not-found-err
+        (err code)))))
+
+;; see operable-trait
+(define-public (set-approved (nftIndex uint) (operator principal) (approved bool))
```

Comment:
The function should allow to set-approved for nfts owned at any time, not just right now.